flask_resources/parsers/base.py

```diff
@@ -52,49 +48,64 @@ def inner(self, *args, **kwargs):
     return decorator
 
 
-class ResourceFlaskParser(FlaskParser):
-    """Customized FlaskParser for our needs."""
-
-    DEFAULT_VALIDATION_STATUS = 400
+class _ExcludeSchema(ma.Schema):
+    class Meta:
+        unknown = ma.EXCLUDE
 
 
-class RequestParser(object):
+class RequestParser:
     """Request parser.
 
-    The class is defining the following attributes:
-    str location: Where on the request to search for values.
-        Can include one of ``('json', 'querystring', 'form',
-        'headers', 'cookies', 'files')``.
-    raw_args_fn: Function to return raw arguments from the current request
-        object being parsed.
+    Base class for parsing request.
     """
 
-    location = None
-    raw_args_fn = _raw_args
+    DEFAULT_SCHEMA_CLASS = _ExcludeSchema
+    """The schema class to use if a dictionary is passed."""
 
-    def __init__(self, argmap=None, allow_unknown=True):
+    def __init__(self, schema=None, allow_unknown=None):
         """Constructor.
 
-        :param argmap: Either a `marshmallow.Schema`, a `dict`
-            of argname -> `marshmallow.fields.Field` pairs, or a callable
-            which accepts a request and returns a `marshmallow.Schema`.
-        :params bool allow_unknown: Allow unknown fields i.e not declared to
-            the `argsmap`, to be parsed.
+        :param schema: A marshmallow schema class or a mapping from keys to
+        fields.
         """
-        self.argmap = argmap or {}
+        self._schema = schema
         self.allow_unknown = allow_unknown
+        if allow_unknown is not None:
+            warnings.warn(
+                "The allow_unknown keyword argument is deprecated and has no "
+                "effect. Use unknown values control on marshmallow instead.",
+                DeprecationWarning,
+            )
+
+    @property
+    def schema(self):
+        """Build the schena class."""
+        if self._schema is None:
+            return None
+        if isinstance(self._schema, dict):
+            cls_ = self.DEFAULT_SCHEMA_CLASS
+            if self.allow_unknown is True:
+
+                class _IncludeSchema(cls_):
+                    class Meta:
+                        unknown = ma.INCLUDE
+
+                cls_ = _IncludeSchema
+
+            return cls_.from_dict(self._schema)()
```

Comment:
That's a neat technique: dynamic creation of Schema class for Meta **with** another layer of dynamic Schema creation for fields. I'll add that to my toolbelt thanks!

Perhaps a comment here to tie this to the deprecation warning, so that future developers know to remove it along with the deprecation warning when we completely remove `allow_unknown`.

Reply:
👍 Yep, makes sense. I've added a comment above now.